Library/Homebrew/requirements/perl_requirement.rb

```diff
@@ -0,0 +1,28 @@
+class PerlRequirement < Requirement
+  fatal true
+  default_formula "perl"
+
+  def initialize(tags)
+    @version = tags.shift if /(\d.\d{1,2})/ === tags.first
+    raise "PerlRequirement requires a version!" unless @version
+    super
+  end
+
+  satisfy(:build_env => false) do
+    which_all("perl").detect do |perl|
+      perl_version = /(\d\.\d(\d)?)/.match Utils.popen_read(perl, "--version")
```

Comment:
Yeah, it outputs the same thing entirely as far as I can see. I'm perfectly happy to let my preference for as much simplicity as possible die a little if it makes you happier on robustness here.